src/test/java/org/opentripplanner/middleware/GetMonitoredTripsTest.java

```diff
@@ -0,0 +1,154 @@
+import static org.opentripplanner.middleware.auth.Auth0Connection.isAuthDisabled;
+
+/**
+ * Tests to simulate getting trips as an Otp user with enhanced admin credentials. The following config parameters must
+ * be set in configurations/default/env.yml for these end-to-end tests to run:
+ *
+ * AUTH0_DOMAIN set to a valid Auth0 domain.
+ *
+ * AUTH0_API_CLIENT set to a valid Auth0 application client id.
+ *
+ * AUTH0_API_SECRET set to a valid Auth0 application client secret.
+ *
+ * OTP_API_ROOT set to a live OTP instance (e.g. http://otp-server.example.com/otp).
+ *
+ * OTP_PLAN_ENDPOINT set to a live OTP plan endpoint (e.g. /routers/default/plan).
+ *
+ * The following environment variable must be set for these tests to run: - RUN_E2E=true.
+ *
+ * Auth0 must be correctly configured as described here: https://auth0.com/docs/flows/call-your-api-using-resource-owner-password-flow
+ */
+public class GetMonitoredTripsTest {
+    private static AdminUser adminUser;
+    private static OtpUser otpUser1;
+    private static OtpUser otpUser2;
+
+    /**
+     * Whether tests for this class should run. End to End must be enabled and Auth must NOT be disabled. This should be
+     * evaluated after the middleware application starts up (to ensure default disableAuth value has been applied from
+     * config).
+     */
+    private static boolean testsShouldRun() {
+        return isEndToEnd && !isAuthDisabled();
+    }
+
+    /**
+     * Create Otp and Admin user accounts. Create Auth0 account for just the Otp users. If
+     * an Auth0 account is created for the admin user it will fail because the email address already exists.
+     */
+    @BeforeAll
+    public static void setUp() throws IOException, InterruptedException {
+        // Load config before checking if tests should run.
+        OtpMiddlewareTest.setUp();
+        assumeTrue(testsShouldRun());
+        // Mock the OTP server TODO: Run a live OTP instance?
+        TestUtils.mockOtpServer();
+        String email = String.format("test-%s@example.com", UUID.randomUUID().toString());
+        otpUser1 = PersistenceUtil.createUser(String.format("test-%s@example.com", UUID.randomUUID().toString()));
+        otpUser2 = PersistenceUtil.createUser(email);
+        adminUser = PersistenceUtil.createAdminUser(email);
+        try {
+            User auth0User = Auth0Users.createAuth0UserForEmail(otpUser1.email, TEMP_AUTH0_USER_PASSWORD);
+            otpUser1.auth0UserId = auth0User.getId();
+            Persistence.otpUsers.replace(otpUser1.id, otpUser1);
+            auth0User = Auth0Users.createAuth0UserForEmail(otpUser2.email, TEMP_AUTH0_USER_PASSWORD);
+            otpUser2.auth0UserId = auth0User.getId();
+            Persistence.otpUsers.replace(otpUser2.id, otpUser2);
+            // Uncommenting will fail set-up because the email address already exists with Auth0.
+//            auth0User = createAuth0UserForEmail(otpUser.email, TEMP_AUTH0_USER_PASSWORD);
```

Comment:
Is this comment still needed?

Reply:
It was to highlight the fact that if both the otp and admin users are created fully they would have an account with Auth0. Only here it would fail because the email already exists with Auth0. In other words the only way I can see both an otp and admin user having the same email adddress is if one isn't authenticated with Auth0. Is it possible for them to be authenticated with different Auth0 tenants though???

Reply:
The correct method to use would have been Auth0User.createNewAuth0User but this and subsequent methods would require altering to accomodate testing. I've continued to use Auth0User.createNewAuth0User and updated the comments.